src/coreclr/vm/ceeload.cpp

```diff
@@ -292,13 +301,15 @@ void Module::NotifyProfilerLoadFinished(HRESULT hr)
                     g_profControlBlock.pProfInterface->ModuleAttachedToAssembly((ModuleID) this,
                                                                                 (AssemblyID)m_pAssembly);
                 }
+
+                profilerCallbackHappened = TRUE;
             }
             END_PIN_PROFILER();
         }
 
         // If there are more types than before, add these new types to the
         // assembly
-        if (!IsResource())
+        if (!IsResource() && profilerCallbackHappened)
```

Comment:
This is great.  Can we also reverse the order of these checks since I'd expect `profilerCallbackHappened == false` most often?

```suggestion
        if (profilerCallbackHappened && !IsResource())
```